### DISCUSSION TITLE
Documentation for `sync` option in `derive()`

### DISCUSSION THREAD

**Discussion by hastebrot** [pmndrs/valtio#851] [General]
I wondered what the `sync` option in `derive()` does. I looked into the source code and found that with `sync: true` the proxy value updates will be synced immediately.

```js
if (notifyInSync) {
  unmarkPending(sourceObject)
} else {
  subscription.p = Promise.resolve().then(() => {
    delete subscription.p // promise
    unmarkPending(sourceObject)
  })
}
```

I wonder if there can be some documentation about this option. For my unittests I had to use `await tick();` before I can make assertions on the changed proxy/derive values, before I found the `sync` option which allows to skip this step.

```ts
export const tick = () => {
  // schedules a function to be executed in the next iteration of the event loop,
  // after the current event loop cycle completes.
  return new Promise((resolve) => {
    setImmediate(resolve);
  });
};
```

**Comment by dai-shi** [comment]
First of all, you are welcome to open a PR to improve docs. It has been improved like that way.

`sync: true` in derive() is notifyInSync for subscribe. It's technically how it works. However, usage-wise, `sync: true` behaves somewhat unexpectedly. So, I'm not sure if it's something useful for many cases. `await tick()` seems correct in your case.
